#! /bin/sh /usr/share/dpatch/dpatch-run
## 52_dshutdown_links.dpatch by Petter Reinholdtsen

Make sure to not not insert unwanted stop symlinks for scripts that
are already configured.  BTS issue #464017.

@DPATCH@
--- insserv-1.11.0.orig/insserv.c
+++ insserv-1.11.0/insserv.c
@@ -2582,11 +2582,14 @@
 	     * default_stop arn't used in SuSE Linux.
 	     */
 	    if (script_inf.default_stop && script_inf.default_stop != empty) {
-		if (service && !del)
-		  {
-		    service->lvlk = str2lvl(script_inf.default_stop);
-		    runlevels(token, script_inf.default_stop);
-		  }
+	        if (service) {
+		    if (service->lvlk) {
+		        runlevels(token, lvl2str(service->lvlk));
+		    } else if (!del) {
+		        service->lvlk = str2lvl(script_inf.default_stop);
+			runlevels(token, script_inf.default_stop);
+		    }
+		}
 	    }
 #endif /* USE_STOP_TAGS */
 	}
